### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by bfutor**
Adding a "Why" and "origin story" per a best practices for README guide - Noticed we were missing them

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +86,27 @@ Hello world!
 flox [my-project] $ exit    # <- Exit environment 💃.
 ```
 
+## ❓ Why
+
+Modern developers build software on top of a dynamic set of tools,
```
Comment (rossturk) [inline README.md:91]:
```suggestion
We all build software on top of a dynamic set of tools,
```

File: README.md [hunk 2]
```diff
@@ -86,6 +86,27 @@ Hello world!
...
+frameworks and packages, allowing them to move quickly and only
```
Comment (rossturk) [inline README.md:92]:
```suggestion
frameworks and packages, allowing us to move quickly and only
```

File: README.md [hunk 3]
```diff
@@ -86,6 +86,27 @@ Hello world!
...
+build what’s necessary. However, each new wave of dev tooling
+innovation results in an entirely new set of dependencies that
+need to be managed. What starts as a simple app or microservice
+quickly grows complex, and developers find themselves dealing
```
Comment (rossturk) [inline README.md:96]:
```suggestion
quickly grows complex, and
```

File: README.md [hunk 4]
```diff
@@ -86,6 +86,27 @@ Hello world!
...
+with the realities of a constantly expanding and diverse supply
```
Comment (rossturk) [inline README.md:97]:
```suggestion
turns into a expanding and fragmented supply
```

File: README.md [hunk 5]
```diff
@@ -86,6 +86,27 @@ Hello world!
...
+chain. Flox brings reproducibility and consistency to complex
+software development lifecycles.
+
+## 📘 Origins
+
+Flox began its life during the deployment of Nix at
+The D. E. Shaw Group, where it quickly proved invaluable
```
Comment (rossturk) [inline README.md:104]:
```suggestion
the D. E. Shaw group, where it quickly proved invaluable
```

**Review by rossturk — CHANGES_REQUESTED** [review]

**Comment by robinbrantley** [comment]
@rossturk - FYI, the text was copied from the new website and the notion blurb doc - just mentioning in case perhaps those should be changed as well?

**Pull request merged**